Cython/Compiler/PyrexTypes.py

```diff
@@ -1039,6 +1069,34 @@ def specialize(self, values):
     def cast_code(self, expr_code):
         return expr_code
 
+    # incref paths are often special-cased for memoryviews. Therefore pass an explicit
+    # flag to make it clear that memoryviewslices should be included. This provides a path
+    # to making the code more generic, without requiring it to be all done at once
+    def generate_incref(self, cname, do_for_memoryviewslice=False, **kwds):
+        if do_for_memoryviewslice:
+            # FIXME use_utility_code
+            return "__PYX_INC_MEMVIEW(&%s, %d);" % (cname, int(kwds['have_gil']))
+
+    # decref however did look to always apply for memoryview slices
+    # with "have_gil" set to True by default
+    def generate_xdecref(self, cname, have_gil=True, **kwds):
+        # FIXME use_utility_code
+        return "__PYX_XDEC_MEMVIEW(&%s, %d);" % (cname, int(have_gil))
+
+    def generate_xdecref_clear(self, cname, clear_before_decref=False,
+                                do_for_memoryviewslice=False, **kwds):
+        return "%s %s.memview = NULL; %s.data = NULL;" % (
+                self.generate_xdecref(cname, do_for_memoryviewslice=True, **kwds),
+                cname, cname)
+
+    def generate_decref_clear(self, cname, **kwds):
+        # memoryviews don't currently distinguish between xdecref and decref
+        return self.generate_xdecref_clear(cname, **kwds)
+
+    # memoryviews don't participate in giveref/gotref
+    generate_gotref = generate_xgotref = generate_xgiveref = generate_giveref = lambda *args: None
```

Comment:
Again: `None` is dangerous here. Better use an empty string.
Also, not accepting `**kwargs` here is at least inconsistent. Why not use the existing dummy methods instead of a new lambda?

Reply:
The existing dummy methods are designed to fail for types with `needs_refcounting` (since if they end up called then they really should exist). I've switched it to an empty string though.